ios/FacebookPlugin.mm

```diff
@@ -282,6 +282,40 @@ - (void) getAuthResponse:(NSDictionary *)opts withRequestId:(NSNumber *)requestI
   [[PluginManager get] dispatchJSResponse:authResponse withError:nil andRequestId:requestId];
 }
 
+- (void) logEvent:(NSDictionary *)opts {
+  NSLOG(@"{facebook} logEvent");
+  NSString     * eventName  = [opts objectForKey:@"eventName"];
+  NSNumber     * valueToSum = [opts objectForKey:@"valueToSum"];
+  NSDictionary * parameters = [opts objectForKey:@"parameters"];
+  // Log the event
+  if(valueToSum != [NSNull null]){
+    if(parameters != [NSNull null]){
+      [FBAppEvents logEvent:eventName valueToSum:[valueToSum doubleValue] parameters:parameters ];
+    } else {
+      [FBAppEvents logEvent:eventName valueToSum:[valueToSum doubleValue] ];
+    }
+  } else {
+    if(parameters != [NSNull null]){
+      [FBAppEvents logEvent:eventName parameters:parameters ];
+    } else {
+      [FBAppEvents logEvent:eventName ];
+    }
+  }
+}
+
+- (void) logPurchase:(NSDictionary *)opts {
+  NSLOG(@"{facebook} logPurchase");
+  NSNumber     *purchaseAmount = [opts objectForKey:@"purchaseAmount"];
+  NSString     *currency       = [opts objectForKey:@"currency"];
+  NSDictionary *parameters     = [opts objectForKey:@"parameters"];
+  // Log the purchase
+  if(parameters != [NSNull null]){
+    [FBAppEvents logPurchase:[purchaseAmount doubleValue] currency:currency];
+  } else {
+    [FBAppEvents logPurchase:[purchaseAmount doubleValue] currency:currency parameters:parameters];
```

Comment:
What about just setting parameters to `nil` if the test passes and then we only need the `[FBAppEvents logPurchase:[purchaseAmount doubleValue] currency:currency parameters:parameters];` call. The shorter version just delegates to this method anyway [link](https://github.com/facebook/facebook-ios-sdk/blob/master/FBSDKCoreKit/FBSDKCoreKit/FBSDKAppEvents.m#L296-L302)